packages/land/contracts/common/ERC721BaseToken.sol

```diff
@@ -195,14 +196,14 @@ abstract contract ERC721BaseToken is Context, IERC721, IERC721Errors, WithSuperO
     /// @param to The address receiving the token.
     /// @param tokenId The token being transferred.
     function _doTransfer(address msgSender, address from, address to, uint256 tokenId) internal {
-        require(to != address(0), "NOT_TO_ZEROADDRESS");
+        if (to == address(0)) {
+            revert InvalidAddress();
+        }
         bool operatorEnabled = _checkFromIsOwner(from, tokenId);
-        require(
-            msgSender == from ||
-                _isApprovedForAllOrSuperOperator(from, msgSender) ||
-                (operatorEnabled && _getOperator(tokenId) == msgSender),
-            "UNAUTHORIZED_TRANSFER"
-        );
+        bool authorized = msgSender == from || _isApprovedForAllOrSuperOperator(from, msgSender);
+        if (!authorized && !(operatorEnabled && _getOperator(tokenId) == msgSender)) {
+            revert ERC721InsufficientApproval(msgSender, tokenId);
+        }
```

Comment:
why did you change the OR ? why not 

```
bool authorized = msgSender == from || _isApprovedForAllOrSuperOperator(from, msgSender) || (operatorEnabled && _getOperator(tokenId) == msgSender);

if (!authorized) etc
```

Reply:
I think the code is more clear with the added variable and better than: `if (!(authorized || (operatorEnabled && _getOperator(tokenId) == msgSender)))`